### ISSUE TITLE
Update request: texlive 2023 → texlive 2024

### ISSUE THREAD

**Issue by nadir-ishiguro** [NixOS/nixpkgs#350953]
- Package name: texlive
- Latest released version: 2024

- Current version on the unstable channel: 2023
- Current version on the stable/release channel: 2023

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

I'm aware that this is no small task and that it's either already in the works or that there's just no available bandwidth. I don't want this to come off as a demand, I'm just curious and guess others might be as well.

**Notify maintainers**

@veprbl @lovek323 @raskin @jwiegley @xworld21

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by xworld21** [comment]
I might have been very stubborn on this... TeX Live does not have releases, so I have been insisting on having frozen releases in Nixpkgs stable as opposed to a random daily snapshot.

Is there consensus on upgrading TeX Live to 2024 in time for 24.11? I can probably make the update quickly. I was able to build 2024.2 some time ago, it shouldn't be hard to port it to Nixpkgs.

PS: I was hoping to have `texlive` and `texliveUnstable`, or say `texlive2023` and `texlive = texlive2024`, but my plans have stalled (partly for personal reasons, partly because making `texlive` ergonomic is quite hard).

**Comment by mirkolenz** [comment]
I think it'd be great to have TeX Live 2024 included in NixOS 24.11. I'd even argue it may be best to use the initial version of the yearly release and skip the continuous updates from ctan. I had some random breakages due to the rolling updates of ctan (could it be that some/most package authors only test them in time for the yearly release?). IIRC nixpkgs currently contains TeX Live 2023 with all ctan package updates applied, and this version for exmaple is not usable for me because the package `acro` is broken. It seems to be fixed in TeX Live 2024 however. Currently, I use TeX Live 2023 from NixOS 23.11 to mitigate this. BTW: Thanks for your work, I have been following your attempt to have multiple versions of TeX Live in nixpkgs closely!

TLDR: I would be so grateful if you could make that update happen for NixOS 24.11 😄

**Comment by veprbl** [comment]
@mirkolenz 
Using -final distributions had origins in availability of tarballs that need to be downloaded. We have only "recently" started using texlive.info and haven't tried to rely on it for release yet. (During update to 2024 it temporarily banned me, possibly because of some request rate restriction.) As far as bugs go, there probably won't be a snapshot that is bug free. If we want to ensure good user experience, moving slower and fixing bugs in individual packages is the way to go. So, if you have a problem with `acro`, please file an issue for that. Having said that, there is no "rule" why we can not do what you propose. Ideally, that would come with some extra maintainer hands to help with that.

Also, feel free to test #351790, you can pin it as (replace `pr/texlive_2024` with commit hash, if you prefer):
```nix
(import (builtins.fetchTarball https://github.com/NixOS/nixpkgs/archive/pr/texlive_2024.tar.gz) {}).texliveFull
```
